js/views/baseVw.js

```diff
@@ -150,6 +150,51 @@ export default class baseVw extends View {
     super.delegateEvents();
   }
 
+  /**
+   * Returns a given jQuery Element or a locally cached version for optimization.
+   * NOTE: Ensure that child views call super.render() in their render function to clear the cache
+   *       since the DOM elements in the cache will then be stale references.
+   * @param {selector} string - The jQuery selector for the Element.
+   *
+   * @return {element} JQuery The element(s) found in the View's Dom or the View's Cache
+   */
+  getCachedElement(selector) {
+    let element;
+
+    // Ensure we have our cached elements map.
+    if (!this._cachedElementMap) {
+      this._cachedElementMap = new Map();
+    }
+
+    // If the cache has the element, we shall use it.
+    if (this._cachedElementMap.has(selector)) {
+      element = this._cachedElementMap.get(selector);
+    } else {
+      // The cache does not not have the element, therefore query with jQuery and cache it.
```

Comment:
Even with your second response to this, I'm glad you brought it up. Now it will be more likely to come to mind.